docs/architecture/adr-048-consensus-fees.md

```diff
@@ -0,0 +1,181 @@
+# ADR 048: Consensus Fees
+
+## Changelog
+
+- Dec 1, 2021: Initial Draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR describes a machanism to maintain the minimal gas price on consensus level.
+
+## Context
+
+Currently, each validator configures it's own `minimal-gas-prices` in `app.yaml`. But setting a proper minimal gas price is critical to protect network from dos attack, and it's hard for all the validators to pick a sensible value, so we proposal to maintain the minimal gas price in consensus level.
+
+Since tendermint 0.35 has supported mempool prioritization, we can take advantage of that to implement more sophisticated gas fee system.
+
+### Compare Different Designs
+
+#### Constant gas fee
+
+##### Positive
+
+- Predictable.
+
+  This makes client behaviour simpler and predictable. And with current system, users are already live with a de facto constant gas price since most validators don't adjust their minimal gas prices.
+
+##### Negative
+
+- Can't include urgent transactions with priority.
```

Comment:
this would be built into the middleware imo. If a chain needs a special tx they would be aware of this and preprogram it